### ISSUE TITLE
Use portal-vue for modal, popover/tooltip, and toast components

### ISSUE THREAD

**Issue by tmorehouse** [bootstrap-vue/bootstrap-vue#2261]
This issue is more of a note to the core team.

Consider using [portal-vue](https://github.com/LinusBorg/portal-vue) for placing modals, tooltips, popovers, and toasts at the bottom of the document. Although we need to wait for release 2.0.0 (as there is a feature needed that will be available in release 2)

I've been working on the new toast feature which needs portal-vue's features (as toasts need to be stacked in the same container), and have found that it will be also very handy for modals (for fixing z-index/stacking issues) and tooltips/popovers.

Currently portal-vue renders placeholder elements for the `portal` component, which can break layout in some situations (hence the need to wait for release 2.0), although there is a temporary workaround that can fix this (but will need to change when 2.0 is released).

**Comment by TitanFighter** [comment]
@tmorehouse FYI, [portal-vue v2](https://github.com/LinusBorg/portal-vue/tree/master) already released.

**Comment by tmorehouse** [comment]
@TitanFighter yeah, working on it this weekend, although we may wait for a patch release of it to come out first before we release it into the wild.

**Comment by tmorehouse** [comment]
@TitanFighter Toasts will be available in v2.0.0-rc.19

**Comment by tmorehouse** [comment]
Modal will be the next to be converted to using Portal-Vue (minimal change required to support it), followed by Tooltips/Popovers (they need the most re-work)

**Comment by stigok** [comment]
It would be great if the documentation was updated to specify exactly which components needs portal to work. Right now it says "Toasts etc". If I only were using e.g. `b-container`, I don't want portal in my build.
https://github.com/bootstrap-vue/bootstrap-vue/blame/9267d1205b25091ecc5ac532f619825ae53f5375/docs/markdown/intro/README.md#L506
